Remove unused and rarely used irrlicht color functions

HybridDog opened the pull request:
SColor.h contains many functions which are unused and/or perform linear
operations on non-linear 8 bit sRGB color values, such as the plus operator and
`SColor::getInterpolated()`, and there is no documentation about missing gamma
correction.
Some of these functions are not called or called only once:
* `getAverage(s16 color)`: Unused
* `SColor::getLightness()`: Unused
* `SColor::getAverage()`: Claims to determine a color's average intensity but
  calculates something significantly different since SColor represents
  non-linear sRGB values.
* `SColor::getInterpolated_quadratic()`: Claims to interpolate between colors
  but uses the sRGB color space, which is neither physically nor perceptually
  linear.
* `SColorf::getInterpolated_quadratic()`: Unused
* `SColorf::setColorComponentValue()`: Unused

Removing or inlining these functions can simplify the code and documenting
gamma-incorrect operations can reduce confusion about what the functions do.

This PR does the following:
* Remove the above-mentioned unused functions
* Inline `SColor::getAverage()` into
  `CIrrDeviceLinux::TextureToMonochromeCursor()`
* Rename `SColor::getLuminance()` into `SColor::getBrightness()` since it does
  not determine a color's luminance but calculates something which differs
  significantly from physical luminance since SColor represents non-linear sRGB
  values.
* Inline `SColor::getInterpolated_quadratic()` into `GameUI::update()`,
  where it is only used for the alpha value calculation for fading
* Document gamma-incorrect behaviour in docstrings

Roadmap goals: 2.1 Rendering/Graphics improvements (general correctness), 2.2 Internal code refactoring (dead code removal)


## To do

This PR is a Ready for Review.


## How to test

Check that compilation succeeds and that the inlined code calculates the same values as before

sfan5 reviewed (approved):
lgtm

SmallJoker commented on src/client/imagesource.cpp:
```diff
@@ -566,7 +566,9 @@ static void apply_hue_saturation(video::IImage *dst, v2u32 dst_pos, v2u32 size,
 			if (colorize) {
-				f32 lum = dst->getPixel(x, y).getLuminance() / 255.0f;
+				video::SColor col = dst->getPixel(x, y);
+				f32 lum = (0.3f * col.getRed() + 0.59f * col.getGreen()
+					+ 0.11f * col.getBlue()) / 255.0f;
```
These magic values are used twice in the code. I don't think there's any benefit in inlining `getLuminance` in particular.

HybridDog replied:
It's back.

appgurueu commented:
I would prefer:

* Remove only unused functions
* Keep the others with `FIXME` comments

I don't think "de-abstracting" just because the implementations are bad is a good idea.

HybridDog commented:
I have changed it so that only functions called at most one time are removed. Inlining `SColor::getInterpolated_quadratic()` into `GameUI::update()` makes the fading code easier to understand in my opinion and `SColor::getAverage()` is called only in `CIrrDeviceLinux::TextureToMonochromeCursor()`, which, as far as I know, is unused in Luanti.
Instead of `FIXME` comments, I have changed comments to mention the gamma-incorrect behaviour and renamed `getLuminance` to `getBrightness`.
